### PR TITLE
chore: simplify further the hyper/xnet code

### PR DESCRIPTION
**Pull request by rumenov**
Use the start_listerner used by the http handler for opening the xnet server socket

### CODE REVIEW

File: rs/http_endpoints/metrics/src/lib.rs [hunk 1]
```diff
@@ -149,11 +149,11 @@ impl MetricsHttpEndpoint {
     fn start_http(&self, address: SocketAddr) {
         // we need to enter the tokio context in order to create the timeout layer and the tcp
         // socket
-        let _enter = self.rt_handle.enter();
 
         let mut addr = "[::]:9090".parse::<SocketAddr>().unwrap();
         addr.set_port(address.port());
-        let tcp_listener = start_tcp_listener(addr);
+        let tcp_listener = start_tcp_listener(addr, &self.rt_handle);
+        let _enter: tokio::runtime::EnterGuard = self.rt_handle.enter();
```
Comment (tthebst) [inline rs/http_endpoints/metrics/src/lib.rs:156]: Is this necessary?

File: rs/http_endpoints/public/src/lib.rs [hunk 2]
```diff
@@ -305,14 +305,13 @@ pub fn start_server(
...
-    let _enter = rt_handle.enter();
     // TODO(OR4-60): temporarily listen on [::] so that we accept both IPv4 and
     // IPv6 connections. This requires net.ipv6.bindv6only = 0. Revert this once
     // we have rolled out IPv6 in prometheus and ic_p8s_service_discovery.
     let mut addr = "[::]:8080".parse::<SocketAddr>().unwrap();
     addr.set_port(listen_addr.port());
-    let tcp_listener = start_tcp_listener(addr);
-
+    let tcp_listener = start_tcp_listener(addr, &rt_handle);
+    let _enter = rt_handle.enter();
```
Comment (tthebst) [inline rs/http_endpoints/public/src/lib.rs:314]: Same

**Review by tthebst — APPROVED** [review]

File: rs/http_endpoints/public/src/lib.rs (hunk 2 above)
Comment (rumenov) [reply rs/http_endpoints/public/src/lib.rs:314]: Yes, because some of the tower layer require the runtime even on construction

**Review by alin-at-dfinity — APPROVED** [review]
LGTM, thank you.

**Pull request merged**